lib/opentelemetry/trace/exporter/http_client.lua

```diff
@@ -28,29 +34,31 @@ function _M.new(address, timeout, headers)
         uri = uri,
         timeout = timeout,
         headers = headers,
+        httpc = httpc,
     }
+
     return setmetatable(self, mt)
 end
 
 function _M.do_request(self, body)
```

Comment:
Can we consider change `do_request(self, body)` to `do_request(self, body, additional_headers)`.
Then call `do_request(body, ["Content-Encoding": "gzip"])` if gzip is enabled.